pkg/topologyinfo/cpus/cpus.go

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ */
+
+package cpus
```

Comment:
I noticed [here](https://github.com/fromanirh/topologyinfo/tree/main/cpus) that you already worked on cpu_test.go. If the implementation is exactly the same, can we add the cpu_test.go as part of this PR too? Just a suggestion :)

Reply:
This is exactly where I am expecting the discussion to happens. `topologyinfo` here is indeed a narrowed-down, copy/pasted (poor's man vendor) clone of https://github.com/fromanirh/topologyinfo/. The problem is exactly this: being a copy/paste we do have a test suite, but it is not integrated at all with the rest of the tests found elsewhere in the codebase.

I have no preference regarding how to address this issue (if at all)

Reply:
Keep me honest, but I thought that [this](https://github.com/swatisehgal/node-feature-discovery/blob/master/Makefile#L103) would take care of executing the tests?

Reply:
absolutely. But please compare [this](https://github.com/swatisehgal/node-feature-discovery/blob/rte/pkg/topologyinfo/numa/numa_test.go) with [this](https://github.com/swatisehgal/node-feature-discovery/blob/rte/pkg/nfd-worker/nfd-worker-internal_test.go).

We can clearly see the tests I added on `topologyinfo` are written with different style, conventions, tools (`sysfs/fake` instead of `mock`s). This is indeed caused by the poor man's vendoring approach I mentioned. Nevertheless `pkg/topologyinfo` feels alien compared to the rest of the codebase.

Do we want to take care of this? and if so, how?

Reply:
I see your point. I am open to suggestions on the best way to move ahead. I see a couple of options here:

1. Add the topologyinfo pkg code in this repo for now (merge this PR) and later add tests that are of the same style as that in the repo (as another PR).  

1. Import pkg [https://github.com/fromanirh/topologyinfo](https://github.com/fromanirh/topologyinfo) now, later update the import path to https://github.com/openshift-kni/topologyinfo

Reply:
I think the latter is better because makes easier for us to pivot to other information-gathering packages (e.g. `ghw`) later on if we feel it's better or if NFD maintainers ask for it. Updating accordingly.